scyllacloud/hostdialer.go

```diff
@@ -96,8 +96,22 @@ func (s *SniHostDialer) dialInitialContactPoint(ctx context.Context, host *gocql
 		return nil, fmt.Errorf("can't get root CA from configuration: %w", err)
 	}
 
-	return s.connect(ctx, s.dialer, host.HostnameAndPort(), &tls.Config{
-		ServerName:         host.Hostname(),
+	if len(s.connConfig.CurrentContext) == 0 {
+		return nil, fmt.Errorf("current context is empty")
+	}
+
+	contextConf, ok := s.connConfig.Contexts[s.connConfig.CurrentContext]
+	if !ok {
+		return nil, fmt.Errorf("context %q does not exists", s.connConfig.CurrentContext)
+	}
+
+	dcConf, ok := s.connConfig.Datacenters[contextConf.DatacenterName]
+	if !ok {
+		return nil, fmt.Errorf("datacenter %q does not exists", contextConf.DatacenterName)
+	}
+
+	return s.connect(ctx, s.dialer, dcConf.Server, &tls.Config{
+		ServerName:         dcConf.NodeDomain,
```

Comment:
sanity check for empty string? (empty will make it default to the hostname)

Reply:
done